### ISSUE TITLE
filesystem recreated if umounted

### ISSUE THREAD

**Issue by pmacdougall** [sous-chefs/filesystem#99]
Using cookbook version 3.0.2:
If a volume is unmounted and chef-client run again, this seems to re-format the volume every time before mounting.

I don't see where [generic_check_cmd](https://github.com/sous-chefs/filesystem/blob/master/resources/default.rb#L161) string gets executed to prevent this.  It's been a while since I've dealt with ruby, but it looks to me like the logic is just checking that `generic_check_cmd` is defined.

It used to work in v2.01 because the `generic_check_cmd` was part of a [not_if](https://github.com/sous-chefs/filesystem/blob/v2.0.1/providers/default.rb#L137) guard, where it would have been executed.

Also not certain the `elsif` logic is correct in the 3.0.2 version, even if the `generic_check_cmd` was the result of the command being executed:
```
    if force
      return if generic_check_cmd && !ignore_existing
    elsif generic_check_cmd && !shell_out("which mkfs.#{fstype}").exitstatus == 0
      return
    end
```

**Comment by ramereth** [comment]
@pmacdougall that's because it doesn't with how this is written unfortunately. Before it was executed being a part of the `not_if` parameter. What I think needs to happen here is `generic_check_cmd` should be a new helper method that uses `shell_out` to properly test this.

**Comment by MarkGibbons** [comment]
#100 should fix this issue.  Thank you for reporting that problem.

**Comment by ramereth** [comment]
@pmacdougall can you please confirm this PR fixes the issue?

**Comment by pmacdougall** [comment]
LGTM

**Comment by ramereth** [comment]
Fix has been released in 3.0.3.
